sturdy/validator/reward.py

```diff
@@ -340,140 +302,105 @@ def get_rewards(
-            allocs[uids[idx]] = {
-                "apy": 0,
-                "allocations": None,
-            }
-        else:
-            allocs[uids[idx]] = {
-                "apy": apys[uids[idx]],
-                "allocations": responses[idx].allocations,
-            }
+    with get_db_connection() as conn:
+        # get assets and pools that are used to benchmark miner
+        # we get the first row entry - we assume that it is the only response from the database
+        try:
+            request_info = get_request_info(conn, request_uid=request_uid)[0]
+            assets_and_pools = json.loads(request_info["assets_and_pools"])
+        except Exception:
+            return ([], {})
 
-            filtered_allocs[uids[idx]] = {
-                "apy": apys[uids[idx]],
-                "allocations": responses[idx].allocations,
-            }
+        # obtain the miner responses for each request
+        miners = get_miner_responses(conn, request_uid=request_uid)
+        bt.logging.debug(f"filtered allocations: {miners}")
+
+    # TODO: see if we can factor this into its own subroutine
+    # if so, do the same with the same one in validator.py
 
-    sorted_filtered_allocs = dict(sorted(filtered_allocs.items(), key=lambda item: item[1]["apy"], reverse=True))
+    pools = assets_and_pools["pools"]
+    new_pools = {}
+    for uid, pool in pools.items():
+        new_pool = PoolFactory.create_pool(
+            pool_type=pool["pool_type"],
+            web3_provider=self.w3,  # type: ignore[]
+            user_address=(pool["user_address"]),  # TODO: is there a cleaner way to do this?
+            contract_address=pool["contract_address"],
+        )
 
-    sorted_apys = dict(sorted(apys.items(), key=lambda item: item[1], reverse=True))
+        # sync pool
+        new_pool.sync(self.w3)
+        new_pools[uid] = new_pool
 
-    sorted_axon_times = dict(sorted(axon_times.items(), key=lambda item: item[1]))
+    assets_and_pools["pools"] = new_pools
 
-    bt.logging.debug(f"sorted apys:\n{sorted_apys}")
-    bt.logging.debug(f"sorted axon times:\n{sorted_axon_times}")
-    bt.logging.debug(f"sorted filtered allocs:\n{sorted_filtered_allocs}")
+    # TODO: this probably needs more work
+    # TODO: would it be better here to use metrics i.e. liquidityIndex for aave pools?
+    # calculate the "adjusted" yields of the allocations
+    for miner in miners:
+        allocations = json.loads(miner["allocation"])["allocations"]
+        extra_metadata = json.loads(request_info["metadata"])
+        miner_uid = miner["miner_uid"]
+        miner_apy = calculate_apy(allocations, assets_and_pools, extra_metadata)
+        miner_axon_time = miner["axon_time"]
 
-    self.sorted_apys = sorted_apys
-    self.sorted_axon_times = sorted_axon_times
+        miner_uids.append(miner_uid)
+        axon_times[miner_uid] = miner_axon_time
+        apys_and_allocations[miner_uid] = {"apy": miner_apy, "allocations": allocations}
 
```

Comment:
@Shr1ftyy  right now the `sorted_apy` attribute of the validator is empty and does not push values to testnet 

```suggestion
self.sorted_apy = dict(sorted({key: value for key, value in apys_and_allocations["apy"].items()}.items(), key=lambda item: item[1], reverse=True))
```